src/rust/include/rust/transaction.h

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2020 The Zcash developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or https://www.opensource.org/licenses/mit-license.php .
+
+#ifndef ZCASH_RUST_INCLUDE_RUST_TRANSACTION_H
+#define ZCASH_RUST_INCLUDE_RUST_TRANSACTION_H
+
+#include <stddef.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct PrecomputedTxParts;
+typedef struct PrecomputedTxParts PrecomputedTxParts;
+
+/// Calculates identifying and authorizing digests for the given transaction.
+///
+/// If either `txid_ret` or `authDigest_ret` is `nullptr`, the corresponding
+/// digest will not be calculated.
+///
+/// Returns `false` if the transaction is invalid.
```

Comment:
Document that each of `txid_ret` and `authDigest_ret` must either be `nullptr` or point to a 32-byte buffer.